# Snapshot report for `language/expressions/property-accessors/index.js`

The actual snapshot is saved in `index.js.snap`.

Generated by [AVA](https://avajs.dev).

## language/expressions/property-accessors/S11.2.1_A1.1.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ❌ Fail | An error occurred while compiling the test with `g++`. |
> | Execute | ⬜ Skip | *This stage did not run.* |

    Error {
      message: `<stdin>:36:77: error: universal character name refers to a control character␊
      if (_test(_strictNotEquals((scope->*"eval").call(make_shared<String>("Number\\u000b.\\u000bPOSITIVE_INFINITY")), scope->*"Number"->*"POSITIVE_INFINITY"))) {␊
                                                                                  ^~~~~~␊
      <stdin>:36:84: error: universal character name refers to a control character␊
      if (_test(_strictNotEquals((scope->*"eval").call(make_shared<String>("Number\\u000b.\\u000bPOSITIVE_INFINITY")), scope->*"Number"->*"POSITIVE_INFINITY"))) {␊
                                                                                         ^~~~~~␊
      <stdin>:60:79: error: universal character name refers to a control character␊
      if (_test(_strictNotEquals((scope->*"eval").call(make_shared<String>("Number\\t\\u000b\\f  \\n\\r      .\\t\\u000b\\f  \\n\\r      POSITIVE_INFINITY")), scope->*"Number"->*"POSITIVE_INFINITY"))) {␊
                                                                                    ^~~~~~␊
      <stdin>:60:103: error: universal character name refers to a control character␊
      if (_test(_strictNotEquals((scope->*"eval").call(make_shared<String>("Number\\t\\u000b\\f  \\n\\r      .\\t\\u000b\\f  \\n\\r      POSITIVE_INFINITY")), scope->*"Number"->*"POSITIVE_INFINITY"))) {␊
                                                                                                       ^~~~~~␊
      4 errors generated.`,
    }

## language/expressions/property-accessors/S11.2.1_A1.2.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ❌ Fail | An error occurred while compiling the test with `g++`. |
> | Execute | ⬜ Skip | *This stage did not run.* |

    Error {
      message: `<stdin>:36:78: error: universal character name refers to a control character␊
      if (_test(_strictNotEquals((scope->*"eval").call(make_shared<String>("Number[\\u000b\\"POSITIVE_INFINITY\\"\\u000b]")), scope->*"Number"->*"POSITIVE_INFINITY"))) {␊
                                                                                   ^~~~~~␊
      <stdin>:36:105: error: universal character name refers to a control character␊
      if (_test(_strictNotEquals((scope->*"eval").call(make_shared<String>("Number[\\u000b\\"POSITIVE_INFINITY\\"\\u000b]")), scope->*"Number"->*"POSITIVE_INFINITY"))) {␊
                                                                                                              ^~~~~~␊
      <stdin>:60:80: error: universal character name refers to a control character␊
      if (_test(_strictNotEquals((scope->*"eval").call(make_shared<String>("Number[\\t\\u000b\\f  \\n\\r      \\"POSITIVE_INFINITY\\"\\t\\u000b\\f  \\n\\r      ]")), scope->*"Number"->*"POSITIVE_INFINITY"))) {␊
                                                                                     ^~~~~~␊
      <stdin>:60:124: error: universal character name refers to a control character␊
      if (_test(_strictNotEquals((scope->*"eval").call(make_shared<String>("Number[\\t\\u000b\\f  \\n\\r      \\"POSITIVE_INFINITY\\"\\t\\u000b\\f  \\n\\r      ]")), scope->*"Number"->*"POSITIVE_INFINITY"))) {␊
                                                                                                                            ^~~~~~␊
      4 errors generated.`,
    }

## language/expressions/property-accessors/S11.2.1_A2.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S11.2.1_A3_T1.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S11.2.1_A3_T2.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ❌ Fail | An error occurred while executing the test program. |

    Error {
      message: 'NotImplementedException: cannot convert a number to an object',
    }

## language/expressions/property-accessors/S11.2.1_A3_T3.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S11.2.1_A3_T4.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S11.2.1_A3_T5.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S11.2.1_A4_T1.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ❌ Fail | An error occurred while compiling the test with `g++`. |
> | Execute | ⬜ Skip | *This stage did not run.* |

    Error {
      message: `<stdin>:33:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"NaN"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#1: typeof (this.NaN) !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:34:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"NaN"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#2: typeof this[\\"NaN\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:35:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Infinity"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#3: typeof this.Infinity !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:36:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Infinity"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#4: typeof this[\\"Infinity\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:37:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"parseInt"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#5: typeof this.parseInt !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:38:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"parseInt"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#6: typeof this[\\"parseInt\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:39:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"parseFloat"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#7: typeof this.parseFloat !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:40:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"parseFloat"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#8: typeof this[\\"parseFloat\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:41:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"isNaN"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#13: typeof this.isNaN !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:42:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"isNaN"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#14: typeof this[\\"isNaN\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:43:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"isFinite"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#15: typeof this.isFinite !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:44:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"isFinite"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#16: typeof this[\\"isFinite\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:45:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Object"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#17: typeof this.Object !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:46:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Object"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#18: typeof this[\\"Object\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:47:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Number"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#19: typeof this.Number !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:48:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Number"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#20: typeof this[\\"Number\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:49:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Function"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#21: typeof this.Function !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:50:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Function"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#22: typeof this[\\"Function\\"] !== \\"undefined\\""));␊
                                      ^␊
      <stdin>:51:33: error: use of undeclared identifier '_this'␊
      if (_test(_strictEquals(_typeof(_this->*"Array"), make_shared<String>("undefined")))) throw _newThrowable(scope->*"Test262Error", make_shared<String>("#23: typeof this.Array !== \\"undefined\\""));␊
                                      ^␊
      fatal error: too many errors emitted, stopping now [-ferror-limit=]␊
      20 errors generated.`,
    }

## language/expressions/property-accessors/S11.2.1_A4_T2.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S11.2.1_A4_T3.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ❌ Fail | An error occurred while executing the test program. |

    Error {
      message: 'Uncaught Test262Error: #5: typeof Function.prototype.length === "number". Actual: undefined',
    }

## language/expressions/property-accessors/S11.2.1_A4_T4.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ❌ Fail | An error occurred while executing the test program. |

    Error {
      message: 'Uncaught ReferenceError: "Array" is not defined',
    }

## language/expressions/property-accessors/S11.2.1_A4_T5.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ❌ Fail | An error occurred while executing the test program. |

    Error {
      message: 'Uncaught Test262Error: #13: typeof String.prototype.charCodeAt === "function". Actual: undefined',
    }

## language/expressions/property-accessors/S11.2.1_A4_T6.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S11.2.1_A4_T7.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ❌ Fail | An error occurred while executing the test program. |

    Error {
      message: 'Uncaught ReferenceError: "Number" is not defined',
    }

## language/expressions/property-accessors/S11.2.1_A4_T8.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ❌ Fail | An error occurred while executing the test program. |

    Error {
      message: 'Uncaught ReferenceError: "Math" is not defined',
    }

## language/expressions/property-accessors/S11.2.1_A4_T9.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ❌ Fail | An error occurred while executing the test program. |

    Error {
      message: 'Uncaught ReferenceError: "Date" is not defined',
    }

## language/expressions/property-accessors/S8.12.3_A1.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S8.12.3_A2.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/S8.12.3_A3.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ✅ Pass |  |
> | Transpile | ✅ Pass |  |
> | Compile | ✅ Pass |  |
> | Execute | ✅ Pass |  |

    'No error'

## language/expressions/property-accessors/non-identifier-name.js

> | Test Stage | Status | Details |
> | :-- | :-- | :-- |
> | Load | ❌ Fail | An error occurred while loading the test from the `test262` folder. |
> | Transpile | ⬜ Skip | *This stage did not run.* |
> | Compile | ⬜ Skip | *This stage did not run.* |
> | Execute | ⬜ Skip | *This stage did not run.* |

    LoaderError {
      message: 'Negative tests, that is, test with the `negative` frontmatter attibute, are not supported at this time.',
    }
